transport/src/main/java/io/netty/channel/DefaultChannelConfig.java

```diff
@@ -302,6 +319,36 @@ public ChannelConfig setRecvByteBufAllocator(RecvByteBufAllocator allocator) {
         return this;
     }
 
+    public boolean isAutoFlush() {
+        return autoFlush == 1;
+    }
+
+    public ChannelConfig setAutoFlush(boolean autoFlush) {
+        // Auto-flush requires a change in API of EventLoop which can not be done in 4.1, so the SingleThreadEventLoop
+        // is the only implementation that supports it.
+        if (autoFlush && channel.isRegistered() && !(channel.eventLoop() instanceof SingleThreadEventLoop)) {
+            throw new UnsupportedOperationException("Auto flush is only supported for channels using "
+                                                    + SingleThreadEventLoop.class.getName() + " eventloop. Found: "
+                                                    + channel.eventLoop().getClass().getName());
+        }
+        final boolean oldAutoFlush = AUTOFLUSH_UPDATER.getAndSet(this, autoFlush? 1 : 0) == 1;
+        if (autoFlush != oldAutoFlush) {
+            autoFlushModified(autoFlush);
+        }
+        return this;
+    }
+
+    /**
+     * Called when {@link #setAutoFlush(boolean)} is called with a value different than what currently exists.
+     *
+     * @param autoFlush Value of {@code autoFlush} after modification.
+     */
+    protected void autoFlushModified(boolean autoFlush) {
+        if (channel instanceof AbstractChannel) {
+            ((AbstractChannel) channel).autoFlushModified(autoFlush);
+        }
```

Comment:
throw an exception if the channel is not of type `AbstractChannel`. This way the user at least knows that he must override the method for handling it in some way.